### PR TITLE
Resources

### PR DESCRIPTION
**Pull request by VictorKoenders**
Suggestion for a more abstract resource system

### CODE REVIEW

File: game/buildings/Building.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using System.Text;
+
+using asshcii.ecs;
+using asshcii.game.components;
+
+namespace asshcii.game.buildings
+{
+    public class Building : Entity
+    {
+        [Obsolete]
```
Comment (projektir) [inline game/buildings/Building.cs]: If it's obsolete, it should just be removed.

File: game/Program.cs [hunk 2]
```diff
@@ -1,38 +1,36 @@
...
+        static void Main(string[] args)
+        {
+            var neptune = new Planet("Neptune");
 
             var refineryAscii = new Ascii(new char[,] { { 'X', ' ', '=' },
                                                         { 'X', ' ', 'X' }});
 
-            var resources = new Resources(50, 100);
-            var refinery = new Building("Refinery", refineryAscii, resources);
+            // TODO: Make pre-defined buildings that automatically add all the components when they're constructed
```
Comment (projektir) [inline game/Program.cs]: Gotta clean up this TODO.

**Comment by VictorKoenders** [comment]
Fixed reviews

File: ecs.test/Component.cs [hunk 3]
```diff
@@ -1,4 +1,7 @@
-namespace asshcii.ecs.test {
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace asshcii.ecs.test
```
Comment (projektir) [inline ecs.test/Component.cs:3]: Same line, please.

File: ecs.test/Entity.cs [hunk 4]
```diff
@@ -1,18 +1,22 @@
 using System;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 
-namespace asshcii.ecs.test {
+namespace asshcii.ecs.test
+{
```
Comment (projektir) [inline ecs.test/Entity.cs:5]: Here too.

File: ecs.test/Entity.cs [hunk 5]
```diff
@@ -1,18 +1,22 @@
...
 
-    class EntityImpl : asshcii.ecs.Entity {
+    class EntityImpl : asshcii.ecs.Entity
```
Comment (projektir) [inline ecs.test/Entity.cs]: -

**Comment by VictorKoenders** [comment]
C#'s (unofficial) coding style is to put brackets always on their own line. Do we want to deviate from this? And in which case? (namespace, class decl, function decl, if/loop statements)

"There are also scientific studies showing that the brace on its own line (indented to the same level as the code, in fact) improves code comprehension despite what people think they think of the aesthetics. It makes it very clear to the reader, visually and instinctively, which code runs in which context."
https://softwareengineering.stackexchange.com/a/237007

**Pull request merged**